builder/vsphere/common/step_boot_command.go

```diff
@@ -68,17 +70,60 @@ func (s *StepBootCommand) Run(ctx context.Context, state multistep.StateBag) mul
 		pauseFn = state.Get("pauseFn").(multistep.DebugPauseFn)
 	}
 
-	port := state.Get("http_port").(int)
+	var ip string
+	var err error
+	port, ok := state.Get("http_port").(int)
+	if !ok {
+		ui.Error("error retrieving 'http_port' from state")
+		return multistep.ActionHalt
+	}
+
 	if port > 0 {
-		ip := state.Get("http_ip").(string)
```

Comment:
☝🏽 why is this logic only relevant if port is is greater than 0?

Reply:
Based on the SDK, this should not be needed because if it's 0 it would not get this far,

However, the SDK does not account for cheking the allowable range? 

- Ports 0 to 1023 are well-known ports, reserved for system or well-known services.
- Port 0 is reserved and cannot be used for normal network communication.
- Registered ports are from 1024 to 49151.
- The remainder of the ports from 49152 to 65535 can be used dynamically by applications.

So I could remove the `if port > 0 {}` and simply allow this to be handled by the SDK or do the following to limit it to between 1024 and 65535

```go
port := state.Get("http_port").(int)
// Check if port is 80 (HTTP reserved) or within the range of 1024 to 65535.
if port == 80 || (port >= 1024 && port <= 65535) {
    ip := state.Get("http_ip").(string)
    s.Ctx.Data = &bootCommandTemplateData{
        IP:   ip,
        Port: port,
        VMName: s.VMName,
    }
    ui.Sayf("HTTP server is working at http://%v:%v/", ip, port)
}
```

Reply:
Resolved in 53f1531.

Reply:
I'm not sure I understand the issue here. By default, SDK will set the port min to 8000 and port max to 9000. If http_port is 0 an available port between that range will be selected. If you wish to change the port range a users can use `http_port_max` and `http_port_min` in their configuration to change the behavior. 

Please let me know if I am missing something here.

Reply:
Ah, so this existed before. It's used as a condition to skip the IP checks for the HTTP server if the port is set.